pkg/lib/files/files.go

```diff
@@ -527,6 +532,22 @@ func IgnoreSpecificFiles(absPaths ...string) IgnoreFn {
 	}
 }
 
+func ErrorOnBigFiles(path string, fi os.FileInfo) (bool, error) {
```

Comment:
I would restructure this function to follow the signature for `GitIgnoreFn` to return (IgnoreFn, error) and it takes in maxFileSize as a parameter.

Reply:
I think we should only return `IgnoreFn` and not `(IgnoreFn, error)`. There shouldn't be any error returned from a function like `ErrorOnBigFilesFn`. As for its parameters, I think we actually need 2 of them: `maxFileSizeBytes` and `maxMemoryUsagePercent`. This way, we also get rid of 2 non-`files.go`-specific variables.